eth/vm/forks/cancun/state.py

```diff
@@ -15,12 +36,143 @@
 from .computation import (
     CancunComputation,
 )
+from .constants import (
+    BLOB_BASE_FEE_UPDATE_FRACTION,
+    BLOB_TX_TYPE,
+    GAS_PER_BLOB,
+    MIN_BLOB_BASE_FEE,
+    VERSIONED_HASH_VERSION_KZG,
+)
+from .transaction_context import (
+    CancunTransactionContext,
+)
+from .transactions import (
+    BlobTransaction,
+)
+
+
+def fake_exponential(
+    factor: int, numerator: int, denominator: int, max_iterations: int = 10000
```

Comment:
Keep an eye on whether this max_iterations ever breaks a test... we exit early so that tests don't hang with long running exponential values